pkg/grpc/service/studyflow_endpoints.go

```diff
@@ -285,9 +287,15 @@ func (s *studyServiceServer) GetAssignedSurveys(ctx context.Context, req *api_ty
 				cs.ProfileId = profileID
 				resp.Surveys = append(resp.Surveys, cs)
 
-				sDef, err := s.studyDBservice.FindCurrentSurveyDef(req.InstanceId, study.Key, cs.SurveyKey, true)
-				if err != nil {
-					logger.Error.Printf("could not retrieve current survey defintions [%s:%s:%s]: %v", req.InstanceId, study.Key, cs.SurveyKey, err)
+				cacheKey := study.Key + cs.SurveyKey
```

Comment:
I think the cache key should be stronger, at the moment it doesn't take into account the InstanceId for example. I would consider using _id as it is guaranteed to be unique and clearly identifies the specific version of the survey.

Reply:
go idea, will do that

Reply:
I think `_id` is not known at the point when I want to find the survey, so I have to go with studyKey and surveyKey - but I'm ok adding the instanceID as well

Reply:
This makes moving the cache outside of the method context easier - as long as it stays inside the context of one call, it is always the same instanceID I believe. If we start moving it outside of the context of this one call, we should think about mechanisms to empty cache / for cache refreshing after a while, or propagate change info when survey is uploaded (but that could be tricky across multiple instance of the service running in parallel - right now the DB is the sync layer between those).
As long as we use it in the same method, I think concurrency issue can be ignored here - update of the survey version during the call is an unlikely event, and even probably irrelevant for the purpose of this method /goal is to fetch survey infos, definition is fetched right when opening the survey)

Reply:
yeah I think for now we can settle for keeping it here with the correct key, other optimizations I feel are outside of the scope this pull request